harvest/api/_base.py

```diff
@@ -85,14 +81,15 @@ def refresh_cred(self):
         """
         debugger.info(f"Refreshing credentials for {type(self).__name__}.")
 
-    def setup(self, interval: Dict, trader=None, trader_main=None) -> None:
+    def setup(self, interval: Dict, trader_main=None) -> None:
```

Comment:
do not rely on the `trader`. This makes apis more complicated and if users only want to use the api.